apps/frontend/src/pages/ViewListing.tsx

```diff
@@ -1,5 +1,128 @@
+import {
+  Button,
+  Card,
+  CardContent,
+  Container,
+  Grid,
+  Link,
+  Typography,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import _axios_instance from "../_axios_instance.tsx";
+import Carousel from "../extra_components/Carousel.tsx";
+import { useStyles } from "../styles/pageStyles.tsx";
+import { colors } from "../styles/colors.tsx";
+
 const ViewListing = () => {
-  return <h1>ViewListing</h1>;
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const { id } = useParams();
+  const [listingObject, setListingObject] = useState({
+    title: "Sample Title",
+    description: "Sample Description",
+    price: 0,
+    seller_profile: { name: "John Smith" },
+    dateCreated: "2024-05-23T15:30:00Z",
+    distance: 4.2,
+    images: [
+      { url: "https://picsum.photos/1200/400" },
+      { url: "https://picsum.photos/1200/600" },
+      { url: "https://picsum.photos/1200/1200" },
+      { url: "https://picsum.photos/600/1200" },
+      { url: "https://picsum.photos/400/1200" },
+    ],
+  });
+
+  // Load the listing from the api given an ID
+  useEffect(() => {
+    _axios_instance
+      .get("/listing/" + id)
+      .then((response) => {
+        setListingObject(response.data);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, [id]);
+
+  // Convert price to string
+  const priceFormatter = new Intl.NumberFormat("en-CA", {
```

Comment:
This and convertDate are both identically defined here and in Chloe's search PR, it might be worth making a utils file for general shared functions like these

Reply:
I like that idea, we're going to work on having a bit more consistency in today's meeting so I'll make sure to bring that up

Reply:
I'll make a seperate PR for this stuff later